pkg/bgpv1/manager/reconcilerv2/advertisements.go

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconcilerv2
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/bgpv1/manager/instance"
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+)
+
+type AdvertisementsReconcilerParams struct {
+	Logger     logrus.FieldLogger
+	Ctx        context.Context
+	Reconciler string
+	Enabled    bool
+
+	Instance     *instance.BGPInstance
+	InstanceName string
+
+	CurrentAdvertisements []*types.Path
+	ToAdvertise           []*types.Path
+}
+
+// ExportAdvertisementsReconciler reconciles the state of the BGP advertisements
+// with the provided toAdvertise list and returns a list of the advertisements
+// currently being announced.
+func ExportAdvertisementsReconciler(params *AdvertisementsReconcilerParams) ([]*types.Path, error) {
+	var (
+		// logger for the reconciler
+		l = params.Logger.WithFields(logrus.Fields{
+			"reconciler": params.Reconciler,
+			"instance":   params.InstanceName,
+		})
+		// holds advertisements which must be advertised
+		toAdvertise []*types.Path
+		// holds advertisements which must remain in place
+		toKeep []*types.Path
+		// holds advertisements which must be removed
+		toWithdraw []*types.Path
+		// the result of advertising toAdvertise.
+		newAdverts []*types.Path
+	)
+
+	l.Debug("begin reconciling advertisements for BGP instance")
```

Comment:
Isn't this a bit too verbose logging, given that we may call this multiple times per reconciler for each reconcile? Maybe we should log only when adding new advertisements / withdrawing paths?